src/python/ImpactX.cpp

```diff
@@ -227,14 +230,32 @@ void init_ImpactX (py::module& m)
             "Enable or disable eigenemittance diagnostic calculations (default: disabled)."
         )
         .def_property("space_charge",
-             [](ImpactX & /* ix */) {
-                 return detail::get_or_throw<bool>("algo", "space_charge");
-             },
-             [](ImpactX & /* ix */, bool const enable) {
-                 amrex::ParmParse pp_algo("algo");
-                 pp_algo.add("space_charge", enable);
-             },
-             "Enable or disable space charge calculations (default: enabled)."
+            [](ImpactX & /* ix */) -> std::string {
+                return detail::get_or_throw<std::string>("algo", "space_charge");
+            },
+            [](ImpactX & /* ix */, std::variant<bool, std::string> space_charge_v) {
+                if (std::holds_alternative<bool>(space_charge_v)) {
+                    // TODO: bool is deprecated since 25.03, remove some time after
+                    amrex::ParmParse pp_algo("algo");
+                    if (std::get<bool>(space_charge_v)) {
+                        py::print("sim.space_charge = True is deprecated, please use space_charge = \"3D\"");
+                        pp_algo.add("space_charge", std::string("3D"));
+                    } else {
+                        py::print("sim.space_charge = False is deprecated, please use space_charge = \"off\" (or skip for default)");
+                        pp_algo.add("space_charge", std::string("off"));
```

Comment:
Yes, I see that the "off" string is supported, so I would keep this in the example input files.

Reply:
Agreed. And we can still use `false` as well.

The problem why I need to internally call this `off` in the enum is that `false` is a C++ keyword.

Reply:
Ah, can name it `False` internally (uppercase)